### PR TITLE
[WIP] Security docs and changelog updates for auth refactor.

### PR DESCRIPTION
**Pull request by Firesphere**

### CODE REVIEW

File: docs/en/04_Changelogs/4.0.0.md [hunk 1]
```diff
@@ -1106,19 +1107,89 @@ specific functions.
 
 The methods `register` and `unregister` on `Authenticator` are deprecated in favor of the `Config` system. This means that any custom Authenticator needs to be registered through the yml config:
 ```yaml
-SilverStripe\Security\Authenticator;
-  authenticators:
-    - MyVendor\MyModule\MyAuthenticator
+SilverStripe\Core\Injector\Injector:
```
Comment (chillu) [inline docs/en/04_Changelogs/4.0.0.md]: Most of this should move to the actual security docs - it's less of an upgrade concern. The changelog is getting far too long with this stuff all inlined.

Comment (Firesphere) [inline docs/en/04_Changelogs/4.0.0.md]: Cool, will do asap :)

**Comment by dhensby** [comment]
@firesphere, a rebate would be better than adding a merge commit and the commits need squashing as the commit messages are not quite optimal at the moment ;)

**Comment by Firesphere** [comment]
@dhensby I'm aware of that, but I didn't have time to fix everything properly today :(

Have discussed with @chillu, I'll try to update/rewrite/change asap

**Pull request closed without merging**